### DISCUSSION TITLE
0.5.2.6

### DISCUSSION THREAD

**Discussion by rkolesnev** [confluentinc/parallel-consumer#616] [General]
# Improvements

* feature: Micrometer metrics (#594)

# Fixes

* fix: Don’t drain mode shutdown kills inflight threads (#559)
* fix: Drain mode shutdown doesn’t pause consumption correctly (#552)
* fix: RunLength offset decoding returns 0 base offset after no-progress commit - related to (#546)
* fix: Transactional PConsumer stuck while rebalancing - related to (#541)

# GH generated:

## What's Changed
* fix: runlength decoding after no-progress commit by @rkolesnev in https://github.com/confluentinc/parallel-consumer/pull/563
* Fix deadlock between pc-control and pc-broker-poll threads where partitions are revoked  by @nachomdo in https://github.com/confluentinc/parallel-consumer/pull/548
* build(deps-dev): bump mockito.version from 4.9.0 to 5.1.1 by @dependabot in https://github.com/confluentinc/parallel-consumer/pull/556
* build(deps): bump vertx.version from 4.3.6 to 4.4.1 by @dependabot in https://github.com/confluentinc/parallel-consumer/pull/566
* build(deps): bump versions-maven-plugin from 2.14.1 to 2.15.0 by @dependabot in https://github.com/confluentinc/parallel-consumer/pull/555
* build(deps-dev): bump threeten-extra from 1.7.1 to 1.7.2 by @dependabot in https://github.com/confluentinc/parallel-consumer/pull/528
* Do not create trace log message when trace logging is not enabled by @lennehendrickx in https://github.com/confluentinc/parallel-consumer/pull/544
* build(deps): bump maven-release-plugin from 3.0.0-M7 to 3.0.1 by @dependabot in https://github.com/confluentinc/parallel-consumer/pull/580
* PL-211: Update dependencies from dependabot, Add mvnw, use mvnw in jenkins by @eddyv in https://github.com/confluentinc/parallel-consumer/pull/583
* PL-211: update dependencies from dependabot by @eddyv in https://github.com/confluentinc/parallel-consumer/pull/589
* add instance id into thread name by @flashmouse in https://github.com/confluentinc/parallel-consumer/pull/578
* workerThreadPool is initialized lazily by @pweso in https://github.com/confluentinc/parallel-consumer/pull/531
* build(deps): bump reactor-core from 3.5.1 to 3.5.7 by @dependabot in https://github.com/confluentinc/parallel-consumer/pull/585
* Pl 175 DRAIN mode fix by @niamhthornbury in https://github.com/confluentinc/parallel-consumer/pull/591
* build(deps): bump junit.version from 5.10.0-M1 to 5.10.0-RC1 by @dependabot in https://github.com/confluentinc/parallel-consumer/pull/598
* build(deps): bump guava from 32.0.1-jre to 32.1.1-jre by @dependabot in https://github.com/confluentinc/parallel-consumer/pull/595
* build(deps): bump vertx.version from 4.4.3 to 4.4.4 by @dependabot in https://github.com/confluentinc/parallel-consumer/pull/590
* build(deps-dev): bump junit.platform.version from 1.10.0-M1 to 1.10.0-RC1 by @dependabot in https://github.com/confluentinc/parallel-consumer/pull/599
* improves: Error in onPartitionsAssigned in parallel consumer by @nachomdo in https://github.com/confluentinc/parallel-consumer/pull/537
* Feature/health metrics micrometer by @rkolesnev in https://github.com/confluentinc/parallel-consumer/pull/613
* fix: example in readme, license headers by @rkolesnev in https://github.com/confluentinc/parallel-consumer/pull/614
* PL-176 handle close dont drain mode gracefully by @rkolesnev in https://github.com/confluentinc/parallel-consumer/pull/615

## New Contributors
* @lennehendrickx made their first contribution in https://github.com/confluentinc/parallel-consumer/pull/544
* @eddyv made their first contribution in https://github.com/confluentinc/parallel-consumer/pull/583
* @flashmouse made their first contribution in https://github.com/confluentinc/parallel-consumer/pull/578
* @pweso made their first contribution in https://github.com/confluentinc/parallel-consumer/pull/531
* @niamhthornbury made their first contribution in https://github.com/confluentinc/parallel-consumer/pull/591

**Full Changelog**: https://github.com/confluentinc/parallel-consumer/compare/0.5.2.5...0.5.2.6

<hr /><em>This discussion was created from the release <a href='https://github.com/confluentinc/parallel-consumer/releases/tag/0.5.2.6'>0.5.2.6</a>.</em>

**Comment by colinkuo** [comment]
@rkolesnev  Thanks for putting together the notes. Perhaps the following improvement is included in 0.5.26?
https://github.com/confluentinc/parallel-consumer/commit/8417498c46944acfcb18ba5ad2992b22011d6c88

**Reply by rkolesnev** [reply]
@colinkuo - yes it is indeed - it's included in the full change list (under GitHub generated) - must have been missed from Changelog - will add it there too.
